### ISSUE TITLE
No tag id being displayed

### ISSUE THREAD

**Issue by lindsay-twoblacksheep** [semlette/nfc_in_flutter#35]
Hi I'm using version 2.0.2 and I am able to read my tag, however I am trying to read the serial number/id but it get returned as an empty string.

Any idea what could be wrong?

**Comment by semlette** [comment]
Is this on Android or iOS?
Also could you tell more about the tags you're using?

**Comment by James-A-White** [comment]
I'm having the same issue.  I'm using an NTAG213, Tag type: ISO 14443-4, running on the latest version of iOS. When I run NFC Tools on the same phone and the same NFC tag, it gives me the serial number. I'd love to be able to read that serial number using NFC_in_flutter.  Thanks!

**Comment by the-white-cat** [comment]
Same issue there with NTAG213 ISO 14443-3A, iOS - no id provided, NFC Tools displays it correctly. Android seems working fine

**Comment by Sreadon** [comment]
Hi, same on my side with NTAG215. On Android it's perfect I can retrieve the tag's ID even if not NDEF formatted but on iOs the retrieved tag's ID is null.

Any ideas of how I could implement something to retrieve the tag's ID on iOs? I don't need to read / write data on the tag, just to read the tag's ID :) Tested on an iPhone 2020 SE.

**Comment by the-white-cat** [comment]
@Sreadon yep, we had to write custom approach for iOS using NFCTagReaderSession. However, author mentioned "tags" branch in a different discussion, but it didn't worked for us:(

**Comment by Sreadon** [comment]
Does your custom solution with NFCTagReaderSession works? May I ask an help on how to implement this if it's the case? I'm a total noob on iOs code.

**Comment by the-white-cat** [comment]
It works for us. The solution is to cast the tag to MiFare tag which contains id data
Its quite simple, here's the link:
https://developer.apple.com/documentation/corenfc/nfcmifaretag

But again, we work with mifare only

**Comment by Sreadon** [comment]
Thanks, I'll give it a try.
